storage.go

```diff
@@ -0,0 +1,355 @@
+package tprstorage
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/microstorage"
+	"github.com/giantswarm/operatorkit/tpr"
+	"k8s.io/apimachinery/pkg/api/errors"
+	apismeta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	api "k8s.io/client-go/pkg/api/v1"
+)
+
+type TPRConfig struct {
+	Name, Version, Description string
+}
+
+type TPOConfig struct {
+	Name, Namespace string
+}
+
+type Config struct {
+	// Dependencies.
+
+	K8sClient kubernetes.Interface
+	Logger    micrologger.Logger
+
+	// Settings.
+
+	// TPR it the third party resource where data objects are stored.
+	TPR TPRConfig
+
+	// TPOName is the third party object used to store data. This
+	// object will be creatd inside a third party resource specified by
+	// TPR. If the object already exists it will be reused. It is
+	// safe to run multiple Storage instances using the same TPO.
+	TPO TPOConfig
+}
+
+func DefaultConfig() Config {
+	return Config{
+		// Dependencies.
+
+		K8sClient: nil, // Required.
+		Logger:    nil, // Required.
+
+		// Settings.
+
+		TPR: TPRConfig{
+			Name:        "tpr-storage.giantswarm.io",
+			Version:     "v1",
+			Description: "Storage data managed by github.com/giantswarm/tprstorage",
+		},
+
+		TPO: TPOConfig{
+			Name:      "", // Required.
+			Namespace: "giantswarm",
+		},
```

Comment:
I thought you wanted to have a `DefaultTPOConfig` function. When `TPOConfig` is only for internal use we should make it private.